File: objective-c/hamming/hamming/hamming/Hamming.m
//
//  Hamming.m
//  hamming
//
//  Created by Chris Sprague on 3/14/15.
//  Copyright (c) 2015 exercism. All rights reserved.
//

#include "Hamming.h"

@implementation Hamming

// Calculate difference between two strings
+(NSUInteger *)compute:(NSString *)strandA against:(NSString *)strandB
{
    NSUInteger *hammingDisatance = 0;
    
    for (NSInteger i = 0; i < strandA.length; i++) {
        unichar nucleotideA = [strandA characterAtIndex:i];
        unichar nucleotideB = [strandB characterAtIndex:i];
        
        if (nucleotideA != nucleotideB) {
            hammingDisatance++;
        }
    }
    
    return hammingDisatance;
}

@end

File: php/hamming/hamming.php
<?php

// Calculate difference between two strings
function distance($strandA, $strandB) {
	$hammingDistance = 0;
	$length = min(strlen($strandA), strlen($strandB));

	for ($i = 0; $i < $length; $i++) {
		$nucleotideA = $strandA{$i};
		$nucleotideB = $strandB{$i};

		if ($nucleotideA != $nucleotideB) {
			$hammingDistance++;
		}
	}

	return $hammingDistance;
}